pyxrf/model/load_data_from_db.py

```diff
@@ -306,24 +356,27 @@ def map_data2D_srx(runid, fpath,
                     dictv[k].append(e.data[k])
             data = pd.DataFrame(dictv, index=np.arange(1, total_len+1)) # need to start with 1
 
+            
+        # Commented by DG: Just use the detector names from .json configuration file. Do not delete commented code.    
         #express3 detector name changes in databroker
-        if xrf_detector_names[0] not in data.keys():
-            xrf_detector_names = ['xs_channel'+str(i) for i in range(1,4)]
-            config_data['xrf_detector'] = xrf_detector_names
+        #if xrf_detector_names[0] not in data.keys():
+        #    xrf_detector_names = ['xs_channel'+str(i) for i in range(1,4)]
+        #    config_data['xrf_detector'] = xrf_detector_names
 
         if output_to_file:
             print('Saving data to hdf file.')
             write_db_to_hdf(fpath, data,
-                            hdr.start.datashape,
+                            #hdr.start.datashape,
+                            datashape,
                             det_list=config_data['xrf_detector'],
                             pos_list=hdr.start.motors,
                             scaler_list=config_data['scaler_list'],
                             fly_type=fly_type,
                             base_val=config_data['base_value'])  #base value shift for ic
             if 'xs2' in hdr.start.detectors:
                 print('Saving data to hdf file for second xspress3 detector.')
-                tmp = fpath.split('.')
-                fpath1 = '.'.join([tmp[0]+'_1', tmp[1]])
+                root, ext = os.path.splitext(fpath)
+                fpath1 = ''.join([root+'_1', ext])
```

Comment:
Can we use f-strings (or `.format()`) here instead?
```suggestion
                fpath1 = f"{root + '_1'}{ext}"
```